### ISSUE TITLE
async_forward_entry_setup deprecation

### ISSUE THREAD

**Issue by glcos** [jpcornil-git/HA-mcp23017#28]
I get this error since upgrading to Home Assistant 2024.7.1

`2024-07-09 21:10:35.506 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'mcp23017' calls async_forward_entry_setup for integration, mcp23017 with title: 0x27:pin 4 ('Node Active':binary_sensor) and entry_id: c4ab65895dd154fd94be2655bdde7da7, which is deprecated and will stop working in Home Assistant 2025.6, await async_forward_entry_setups instead at custom_components/mcp23017/__init__.py, line 104: await hass.config_entries.async_forward_entry_setup(, please create a bug report at https://github.com/jpcornil-git/HA-mcp23017/issues`

Thanks for having a look

**Comment by jpcornil-git** [comment]
Thanks for reporting !
I just committed a code change to address this => could you give it a try ([Enable beta for a repository](https://hacs.xyz/docs/faq/beta/) and and use 65069c5) ?
Cheers,
jpc

**Comment by glcos** [comment]
I no longer see errors related to async_forward_entry_setup in the log.
However now I get this error:

```
2024-07-09 22:44:42.638 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 0x27:pin 4 ('Node Active':binary_sensor) for mcp23017
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/config_entries.py", line 586, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/mcp23017/__init__.py", line 104, in async_setup_entry
    await hass.config_entries.async_forward_entry_setups(
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/config_entries.py", line 2046, in async_forward_entry_setups
    await integration.async_get_platforms(platforms)
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1160, in async_get_platforms
    import_future.result()
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1160, in async_get_platforms
    import_future.result()
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1160, in async_get_platforms
    import_future.result()
  [Previous line repeated 7 more times]
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1130, in async_get_platforms
    await self.hass.async_add_import_executor_job(
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1073, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1247, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/loader.py", line 1279, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'custom_components.mcp23017.b'
```

**Comment by jpcornil-git** [comment]
Thanks for the quick turnaround ... this HA change seems to have more impact than expected !

I'll investigate/test this on my side first and ask you to test again in a few days/after the week-end probably.

**Comment by jpcornil-git** [comment]
Could you try e793882 ? 

I was able to reproduce your results and found the issue with the first fix (forgot to adapt one argument ...) 

Thanks
jpc

**Comment by glcos** [comment]
Thank you for the fix and your work on this excellent integration, it works fine now!

**Comment by jpcornil-git** [comment]
Thanks for confirming it has been resolved (and your kind words, much appreciated 👍)
I just released v1.2.3 with the fix
Cheers,
jpc
